### PR TITLE
Open cases tests

### PR DESCRIPTION
**Pull request by MichaelCullenMadeTech**
Also refactored the not empty and not null tests a bit more cleanly.

Created a not_completed test to ensure that they are not appearing in those results - not as generic but could be reused across multiple streams.

### CODE REVIEW

File: transformations/tests/generic/not_higher_than.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
-{% test not_higher_than(model, column_name, max_value_column) %}
+ {% test not_higher_than(model, column_name, max_value_column) %}
```
Comment (RhysJones125) [inline transformations/tests/generic/not_higher_than.sql:1]: This is maybe a bit picky and feel free to completely ignore, I feel like the name for this is a bit odd and just being `less_than` is simpler

File: transformations/tests/generic/not_null.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{% test not_null(model, column_name) %}
+
+select *
+from {{ model }}
+where {{ column_name }} is null
```
Comment (RhysJones125) [inline transformations/tests/generic/not_null.sql:5]: Should this be `is not null` or should the name be changed?

Comment (RhysJones125) [reply transformations/tests/generic/not_null.sql:5]: or is the test that this comes back as 0?

File: transformations/tests/generic/not_higher_than.sql (hunk 1 above)
Comment (MichaelCullenMadeTech) [reply transformations/tests/generic/not_higher_than.sql:1]: I get it, I feel I am writing tests with the arguments (other than contextual model and column_name).
So it becomes "not higher than this max valued column".

File: transformations/tests/generic/not_null.sql (hunk 2 above)
Comment (RhysJones125) [reply transformations/tests/generic/not_null.sql:5]: Ignore me. Looking at the other tests they're all kinda reversed and the aim is to get no rows returned

Comment (MichaelCullenMadeTech) [reply transformations/tests/generic/not_null.sql:5]: Again, naming tests as an engineer is different to naming tests in DBT.
The SQL this produces is based on 
```
select
      count(*) as failures,
      count(*) != 0 as should_warn,
      count(*) != 0 as should_error
    from (
      
```
So we are checking for the field to be not null. But within that test you assert a failing condition to equal 0.

**Review by RhysJones125 — APPROVED** [review]

**Pull request merged**